### PR TITLE
Split data table background color into separate theme variable

### PR DESCRIPTION
**Pull request by theaquarium**
This will allow greater theme customization and will match the way colors are set up for other components.

### CODE REVIEW

File: src/resources/ha-style.ts [hunk 1]
```diff
@@ -121,6 +121,9 @@ documentContainer.innerHTML = `<custom-style>
       --paper-slider-container-color: var(--slider-bar-color);
       --ha-paper-slider-pin-font-size: 15px;
 
+      /* set data table style */
+      --mdc-data-table-background-color: var(--card-background-color);
```
Comment (iantrich) [inline src/resources/ha-style.ts]:
```suggestion
      --table-background-color: var(--card-background-color);
```

File: src/components/data-table/ha-data-table.ts [hunk 2]
```diff
@@ -427,7 +427,7 @@ export class HaDataTable extends BaseElement {
       }
 
       .mdc-data-table {
-        background-color: var(--card-background-color);
+        background-color: var(--mdc-data-table-background-color);
```
Comment (iantrich) [inline src/components/data-table/ha-data-table.ts]:
```suggestion
        background-color: var(--data-table-background-color);
```

**Comment by iantrich** [comment]
I don't like that we currently expose paper themes and don't want to get into the habit of exposing it for other proprietary elements. We should be more generic and clear.

**Comment by theaquarium** [comment]
Alright, should be good now.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**